solidity/contracts/GasRouter.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+
+import {Router} from "./Router.sol";
+
+abstract contract GasRouter is Router {
+    // ============ Mutable Storage ============
+    mapping(uint32 => uint256) public handleGasOverhead;
+
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when a remote router's handle gas overhead is set.
+     * @param domain The domain of the new router
+     * @param gas The gas amount used by the  of the new router
+     */
+    event HandleGasOverheadSet(uint32 indexed domain, uint256 gas);
+
+    struct GasRouterConfig {
+        uint32 domain;
+        uint256 handleGasOverhead;
+    }
+
+    /**
+     * @notice Batch version of `enrollRemoteRouterWithGas`
+     * @param gasConfigs The array of GasRouterConfig structs
+     */
+    function setGasOverheadConfigs(GasRouterConfig[] calldata gasConfigs)
+        external
+        onlyOwner
+    {
+        for (uint256 i = 0; i < gasConfigs.length; i += 1) {
+            setGasOverheadConfig(gasConfigs[i]);
+        }
+    }
+
+    /**
+     * @notice Enroll a remote router with a handle gas overhead
+     * @param gasConfig The GasRouterConfig struct
+     */
+    function setGasOverheadConfig(GasRouterConfig calldata gasConfig)
+        public
+        onlyOwner
+    {
+        _setGasOverhead(gasConfig.domain, gasConfig.handleGasOverhead);
+    }
+
+    function setGasOverhead(uint32 domain, uint256 gas) external onlyOwner {
```

Comment:
I also know that i have been advocating for non-struct versions of calls, but i wonder if this is a bit superflous and just adding to the bytecode